strapi-5@0.1.0 error: Cannot find module '@strapi/email/package.json' on first start

## Bug report

### Required System information
- Node.js version: v20.12.2
- NPM version: 10.5.2
- Strapi version: 5.0.0-beta.7
- Database: n/a
- Operating system: macOS 13.6.4
- Is your project Javascript or Typescript: n/a
### Describe the bug

'npm run develop' returns fatal error on first run after fresh installation.

> strapi-5@0.1.0 develop
> strapi develop

⠋ Loading Strapi⠋ Building build context
[INFO] Including the following ENV variables as part of the JS bundle:
    - ADMIN_PATH
    - STRAPI_ADMIN_BACKEND_URL
    - STRAPI_TELEMETRY_DISABLED
✔ Building build context (47ms)
✔ Creating admin (378ms)
[ERROR]  There seems to be an unexpected error, try again with --debug for more information
`                                                                               
│   Error: Cannot find module '@strapi/email/package.json' 
│   Require stack:  
│   - /Users/dev/strapi5/node_modules/@strapi/core/dist/loaders/plugins/get-enabled-plugins.js 
│   - /Users/dev/strapi5/node_modules/@strapi/core/dist/loaders/plugins/index.js
│   - /Users/dev/strapi5/node_modules/@strapi/core/dist/loaders/index.js 
│   - /Users/dev/strapi5/node_modules/@strapi/core/dist/providers/registries.js 
│   - /Users/dev/strapi5/node_modules/@strapi/core/dist/providers/index.js 
│   - /Users/dev/strapi5/node_modules/@strapi/core/dist/Strapi.js 
│   - /Users/dev/strapi5/node_modules/@strapi/core/dist/index.js 
│   - /Users/dev/strapi5/node_modules/@strapi/strapi/dist/cli/commands/admin/create-user.js 
│   - /Users/dev/strapi5/node_modules/@strapi/strapi/dist/cli/commands/index.js 
│   - /Users/dev/strapi5/node_modules/@strapi/strapi/dist/cli/index.js 
│   - /Users/dev/strapi5/node_modules/@strapi/strapi/bin/strapi.js 
│   at Module._resolveFilename (node:internal/modules/cjs/loader:1143:15)           
│   at Module._load (node:internal/modules/cjs/loader:984:27)                       
│   at Module.require (node:internal/modules/cjs/loader:1231:19)                    
│   at require (node:internal/modules/helpers:179:18)
│   at Module.getEnabledPlugins (/Users/dev/strapi5/node_modules/@strapi/core/dist/loaders/plugins/get-enabled-plugins.js:55:25) 
│   at loadPlugins (/Users/dev/strapi5/node_modules/@strapi/core/dist/loaders/plugins/index.js:76:50) 
│   at Module.loadApplicationContext (/Users/dev/strapi5/node_modules/@strapi/core/dist/loaders/index.js:16:5)
│   at Object.register (/Users/dev/strapi5/node_modules/@strapi/core/dist/providers/registries.js:26:17) 
│   at Strapi.register (/Users/dev/strapi5/node_modules/@strapi/core/dist/Strapi.js:301:32) 
│   at Strapi.load (/Users/dev/strapi5/node_modules/@strapi/core/dist/Strapi.js:293:16) 

`
### Steps to reproduce the behavior

npx create-strapi-app@beta my-project  --quickstart
cd my-project
npm run develop

I am facing a similar issue.

`Error: Cannot find module '@strapi\email\package.json'`

- Using TS
- Using PostgreSQL
- Node v20.10.0
- npm v10.8.2

Running `npm ls @strapi/email` yields

4wd-strapi@0.1.0 D:\strapi-test
└─┬ @strapi/strapi@5.0.0-rc.6
  └── @strapi/email@5.0.0-rc.6

> Hey, thanks for the quick reply, here's what I've tried
> 
> ```
> $ node -v
> v20.12.2
> 
> $ npm -v
> 10.5.0
> ```
> 
> 1. `npx create-strapi-app@4.24.4 app`
> 2. `cd app`
> 3. `npm run develop` ✔️
> 4. `npx @strapi/upgrade minor`
> 5. `npx @strapi/upgrade major`
> 6. `npm run develop` ✔️
> 
> Everything worked on my side.
> 
> Not sure what other configuration could cause everything to fail 🤔

Here is my exact error, if that helps
```
[ERROR]  There seems to be an unexpected error, try again with --debug for more information 

┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                                                                                                                  │
│   Error: @strapi/email/package.json couldn't be resolved                                                                                         │
│       at toDetailedDeclaration (/Users/michael/code/strapi-cms/node_modules/@strapi/core/dist/loaders/plugins/get-enabled-plugins.js:65:17)      │
│       at Module.getEnabledPlugins (/Users/michael/code/strapi-cms/node_modules/@strapi/core/dist/loaders/plugins/get-enabled-plugins.js:83:10)   │
│       at loadPlugins (/Users/michael/code/strapi-cms/node_modules/@strapi/core/dist/loaders/plugins/index.js:96:50)                              │
│       at Module.loadApplicationContext (/Users/michael/code/strapi-cms/node_modules/@strapi/core/dist/loaders/index.js:16:5)                     │
│       at Object.register (/Users/michael/code/strapi-cms/node_modules/@strapi/core/dist/providers/registries.js:26:17)                           │
│       at Strapi.register (/Users/michael/code/strapi-cms/node_modules/@strapi/core/dist/Strapi.js:322:32)                                        │
│       at Strapi.load (/Users/michael/code/strapi-cms/node_modules/@strapi/core/dist/Strapi.js:314:16)                                            │
│       at Module.develop (/Users/michael/code/strapi-cms/node_modules/@strapi/strapi/dist/node/develop.js:177:41)                                 │
│       at async action (/Users/michael/code/strapi-cms/node_modules/@strapi/strapi/dist/cli/commands/develop.js:18:5)                             │
│       at async Command.parseAsync (/Users/michael/code/strapi-cms/node_modules/commander/lib/command.js:923:5)                                   │
│                                                                                                                                                  │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

> npm config ls -l | grep "install-strategy"

`install-strategy = "hoisted" `

I get this error when I try to run `npm run develop`.

I even tried following your steps, but I get the following issue:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: strapi-4@0.1.0
npm ERR! Found: @strapi/plugin-cloud@4.25.12
npm ERR! node_modules/@strapi/plugin-cloud
npm ERR!   @strapi/plugin-cloud@"5.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! @strapi/plugin-cloud@"5.0.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: @strapi/strapi@5.0.0
npm ERR! node_modules/@strapi/strapi
npm ERR!   peer @strapi/strapi@"^5.0.0 || ^5.0.0-beta || ^5.0.0-alpha || ^5.0.0-rc" from @strapi/plugin-cloud@5.0.0
npm ERR!   node_modules/@strapi/plugin-cloud
npm ERR!     @strapi/plugin-cloud@"5.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /Users/michael/.npm/_logs/2024-09-26T09_50_52_479Z-eresolve-report.txt
```

I had to do the following steps:

npx create-strapi-app@4.24.4 app
cd app
npm run develop ✔️
npx @strapi/upgrade minor
npx @strapi/upgrade major
rm -rf node_modules/
rm package-lock.json
npm install
npm run develop ✔️

And the outcome is the same:

`[ERROR]  There seems to be an unexpected error, try again with --debug for more information 

┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                                                                                                                         │
│   Error: @strapi/email/package.json couldn't be resolved                                                                                                │
│       at toDetailedDeclaration (/Users/michael/code/next-app/strapi-4/node_modules/@strapi/core/dist/loaders/plugins/get-enabled-plugins.js:65:17)      │
│       at Module.getEnabledPlugins (/Users/michael/code/next-app/strapi-4/node_modules/@strapi/core/dist/loaders/plugins/get-enabled-plugins.js:83:10)   │
│       at loadPlugins (/Users/michael/code/next-app/strapi-4/node_modules/@strapi/core/dist/loaders/plugins/index.js:96:50)                              │
│       at Module.loadApplicationContext (/Users/michael/code/next-app/strapi-4/node_modules/@strapi/core/dist/loaders/index.js:16:5)                     │
│       at Object.register (/Users/michael/code/next-app/strapi-4/node_modules/@strapi/core/dist/providers/registries.js:26:17)                           │
│       at Strapi.register (/Users/michael/code/next-app/strapi-4/node_modules/@strapi/core/dist/Strapi.js:322:32)                                        │
│       at Strapi.load (/Users/michael/code/next-app/strapi-4/node_modules/@strapi/core/dist/Strapi.js:314:16)                                            │
│       at Module.develop (/Users/michael/code/next-app/strapi-4/node_modules/@strapi/strapi/dist/node/develop.js:177:41)                                 │
│       at async action (/Users/michael/code/next-app/strapi-4/node_modules/@strapi/strapi/dist/cli/commands/develop.js:18:5)                             │
│       at async Command.parseAsync (/Users/michael/code/next-app/strapi-4/node_modules/commander/lib/command.js:923:5)                                   │
│                                                                                                                                                         │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘`

I'm facing the same error after upgrading to 5.0.1 from 4.25.12.

My actions:
1. npx @strapi/upgrade major
On updating dependencies the script failed with a version incompatibility error.
2. Manually deleted node_modules and package-lock.json.
3. npm install
Installation completed, but with a warning:
```
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: msw@1.3.0
npm warn Found: typescript@5.3.2
npm warn node_modules/typescript
npm warn   peerOptional typescript@"^4.7 || 5" from @formatjs/intl@2.10.0
npm warn   node_modules/@formatjs/intl
npm warn     @formatjs/intl@"2.10.0" from react-intl@6.6.2
npm warn     node_modules/react-intl
npm warn   7 more (@strapi/admin, @strapi/strapi, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peerOptional typescript@">= 4.4.x <= 5.2.x" from msw@1.3.0
npm warn node_modules/@strapi/admin/node_modules/msw
npm warn   node_modules/@strapi/admin
npm warn
npm warn Conflicting peer dependency: typescript@5.2.2
npm warn node_modules/typescript
npm warn   peerOptional typescript@">= 4.4.x <= 5.2.x" from msw@1.3.0
npm warn   node_modules/@strapi/admin/node_modules/msw
npm warn     msw@"1.3.0" from @strapi/admin@5.0.0
npm warn     node_modules/@strapi/admin
```
4. npx @strapi/upgrade minor
Successful, but with the same warning from step 3.
6. npm run develop
`Error: @strapi\email\package.json couldn't be resolved `

Manual installation of the package helped: `npm install @strapi/email`

`install-strategy = "hoisted"`
Windows, node v20.11.0, npm 10.8.3

--- a/packages/core/core/src/loaders/plugins/get-enabled-plugins.ts
+++ b/packages/core/core/src/loaders/plugins/get-enabled-plugins.ts
@@ -101,7 +101,7 @@ export const getEnabledPlugins = async (strapi: Core.Strapi, { client } = { clie
 
     validatePluginName(packageInfo.strapi.name);
     internalPlugins[packageInfo.strapi.name] = {
-      ...toDetailedDeclaration({ enabled: true, resolve: packagePath, isModule: client }),
+      ...toDetailedDeclaration({ enabled: true, resolve: packageModulePath, isModule: client }),
       info: packageInfo.strapi,
       packageInfo,
     };

